yesod/Yesod/Default/Config.hs

```diff
@@ -232,9 +232,10 @@ withYamlEnvironment :: Show e
                     -> (Value -> Parser a) -- ^ what to do with the mapping
                     -> IO a
 withYamlEnvironment fp env f = do
-    mval <- decodeFile fp
+    mval <- decodeFileEither fp
     case mval of
-        Nothing -> fail $ "Invalid YAML file: " ++ show fp
-        Just (Object obj)
+        Left err ->
+          fail $ "Invalid YAML file: " <> show fp <> " " <> prettyPrintParseException err
```

Comment:
It looks like tests are failing on old versions of GHC/base, where `<>` isn't in scope

```
yesod                            > /home/vsts/work/1/s/yesod/Yesod/Default/Config.hs:238:40: error:
yesod                            >     • Variable not in scope: (<>) :: m0 a0 -> String -> t1
yesod                            >     • Perhaps you meant one of these:
yesod                            >         ‘<$>’ (imported from Prelude), ‘*>’ (imported from Prelude),
yesod                            >         ‘<$’ (imported from Prelude)
yesod                            >     |
yesod                            > 238 |           fail $ "Invalid YAML file: " <> show fp <> " " <> prettyPrintParseException err

```

Reply:
Oh my bad. I fixed it but how did you run those tests? By running `stack test` I didn't see any failures.

Reply:
I didn't run them, that was continuous integration. It tests against a bunch of different configurations. I don't have exact instructions, but you could copy what CI does to test against different configurations (probably just changing the resolver in the stack file)